version: '3'

services:
  vaultwarden:
    restart: unless-stopped
    container_name: vaultwarden
    image: vaultwarden/server:1.28.1-alpine
    volumes:
      - ./data/:/data/
    networks:
      - proxy
    ports:
      - 8062:80
    environment:
      - TZ=Europe/Copenhagen
      - LOGIN_RATELIMIT_MAX_BURST=10
      - LOGIN_RATELIMIT_SECONDS=60
      - DOMAIN=https://vaultwarden.wraaath.com
      - INVITATIONS_ALLOWED=true
      - ADMIN_TOKEN=${ADMIN_TOKEN}
      - EMERGENCY_ACCESS_ALLOWED=true
      - SENDS_ALLOWED=true
      - WEB_VAULT_ENABLED=true
    labels:
        - traefik.enable=true
        - traefik.docker.network=proxy
        - traefik.http.routers.vaultwarden.rule=Host(`vaultwarden.wraaath.com`)
        - traefik.http.routers.vaultwarden.service=vaultwarden
        - traefik.http.services.vaultwarden.loadbalancer.server.port=80
networks:
  proxy:
    external: true
